industry_benchmarks/utils/results_cleanup.py

```diff
@@ -0,0 +1,349 @@
+    """
+    Cleans up the results in the given JSON files.
+
+    Note:
+    To load in the structural analysis data
+
+    >>> loaded = np.load("structural_analysis_data.npz")
+    >>> loaded["time_ps"], loaded["protein_RMSD"], loaded["ligand_RMSD"], loaded["ligand_wander"], loaded["protein_2D_RMSD"]
+
+    Parameters
+    ----------
+    json_files : List[str]
+        List of paths to JSON files to clean up.
+    """
+    for json_file in tqdm(json_files):
+        if not os.path.exists(json_file):
+            print(f"Error: {json_file} does not exist.")
+            continue
+        try:
+            print(f"Working on file {json_file}")
+            backup_file = make_backup(json_file)
+            with open(json_file, "r") as f:
+                results = json.load(f)
+
+            # Check to see if we have already cleaned  up this result
+            result_key = next(k for k in results["protocol_result"]["data"].keys())
+            if (
+                "structural_analysis"
+                in results["protocol_result"]["data"][result_key][0]["outputs"]
+            ):
+                print("Cleaning up file")
+            else:
+                print("Skipping file, already cleaned")
+                continue
+
+            # Check to make sure we don't have more than one proto result
+            # We might have ProtocolUnitResult-* and ProtocolUnitFailure-*
+            # We only handle the case where we have one ProtocolUnitResult
+            protocol_unit_result_count = len(
+                [
+                    k
+                    for k in results["unit_results"].keys()
+                    if k.startswith("ProtocolUnitResult")
+                ]
+            )
+
+            # Check to make sure we don't just have failures
+            # if all failures, tell user to re-run
+            if protocol_unit_result_count != 1:
+                print("More than one ProtocolUnitResult, skipping")
+                continue
+            elif protocol_unit_result_count == 0:
+                print("All protocol units failed, skipping")
+                continue
+
+            # get the name of the key which is a gufe token
+            # for the only ProtocolUnitResult-* in unit_results
+            # this means we can grab the first that matches since there is only
+            # one ProtocolUnitResult-*
+            proto_key = next(
+                k
+                for k in results["unit_results"].keys()
+                if k.startswith("ProtocolUnitResult")
+            )
+
+            results_dir = (
+                Path(results["unit_results"][proto_key]["outputs"]["nc"]["path"])
+                .resolve()
+                .parent
+            )
+            # if the dir doesn't exist, we should try and fix it
+            if not results_dir.is_dir():
+                print("Fixing path to results dir")
+                results_dir = remove_first_reversed_sequential_duplicate_from_path(
```

Comment:
I'm not really sure I understand what this is doing, do you have a quick one or two liner on this?

Reply:
I will add a code comment, its to fix this problem where the cleanup script is in a different relative location to where `openfe quick` was ran *and* a relative path was used. For example:

If out directory looked like:
```
/top/submission/results/lig1-lig2-solvent/data.bin
/top/submission/results/lig1-lig2-solvent-results.json
```

and `openfe quickrun` was executed in the `submission` directory like `openfe quickrun -o results/lig1-lig2-solvent-results.json` then if you ran this script in the `results` folder like `python results_cleanup.py *.json` the `lig1-lig2-solvent-results.json` would have the path to the `data.bin` saved like `results/lig1-lig2-solvent/data.bin` BUT since we are in the results folder, it can't find the file -- we can call `absolute` on the path object to help us out, but the path will look like `/top/submission/results/results/lig1-lig2-solvent/data.bin` (note that `results` is repeated twice). So to fix this, we just remove the "extra" `results` folder. I wanted to support the case where someone might have duplicate dirs higher up (since why not?) so this function walks the full path backward and removes an extra folder if one folder above it has the same name, then stops.

Reply:
done